examples/custom-transport.rs

```diff
@@ -0,0 +1,201 @@
+#![warn(
+    clippy::all,
+    //clippy::missing_docs_in_private_items,
+    clippy::nursery,
+    clippy::pedantic,
+    missing_docs
+)]
+// stable clippy seems to have an issue with await
+#![allow(clippy::used_underscore_binding)]
+
+//!
+
+use anyhow::{anyhow, bail, Result};
+use parking_lot::{Condvar, Mutex};
+use reqwest::Client;
+use sentry::{
+    http::Method, Dsn, Event, Options, RawEnvelope, Request, Transport as SentryTransport,
+    TransportShutdown,
+};
+use sentry_contrib_native as sentry;
+use std::{slice, str::FromStr, sync::Arc, time::Duration};
+use tokio::{
+    runtime::Handle,
+    sync::mpsc::{self, Sender},
+    task,
+};
+
+struct TransportState {
+    sender: Sender<RawEnvelope>,
+    shutdown: Arc<(Mutex<()>, Condvar)>,
+}
+
+async fn send_sentry_request(client: &Client, req: Request) -> Result<()> {
+    let (parts, body) = req.into_parts();
+    let uri = parts.uri.to_string();
+
+    // Sentry should only give us POST requests to send
+    if parts.method != Method::POST {
+        bail!(
+            "Sentry SDK is trying to send an unexpected request of '{}'",
+            parts.method
+        )
+    }
+
+    let rb = client.post(&uri);
+
+    // we cheat so that we don't have to copy all of the bytes of the body
+    // into a new buffer, but we have to fake that the slice is static, which
+    // should be ok since we only need that buffer until the request is finished
+    #[allow(unsafe_code)]
+    let buffer = unsafe {
+        let buf = body.as_bytes();
+        slice::from_raw_parts::<'static, _>(buf.as_ptr(), buf.len())
+    };
+
+    let response = rb
+        .headers(parts.headers)
+        .body(buffer)
+        .send()
+        .await
+        .map_err(|e| anyhow!("Failed to send Sentry request: {}", e))?;
+
+    response
+        .error_for_status()
+        .map_err(|e| anyhow!("Received error response from Sentry: {}", e))?;
+    Ok(())
+}
+
+// we can implement our own transport for Sentry data so that we don't pull in
+// C dependencies (COUGH OPENSSL COUGH) that we don't want
+struct Transport {
+    /// we don't currently use custom certs or proxies, so we can just use the
+    /// same client that the rest of ark uses, if we do start doing that, we
+    /// would need to build the client based on the options during startup()
```

Comment:
> so we can just use the same client that the rest of **ark** uses

What is "ark"?

Reply:
Oops, I just copied the implementation I used in our internal project and accidentally left the comment, you can just delete it.

Reply:
Alright!